mtcpclient/calculate.go

```diff
@@ -39,13 +45,19 @@ func (gc GroupOfConnections) calculateMetricsReport(status tcpclient.ConnectionS
 }
 
 func (gc GroupOfConnections) calculateStdDev(status tcpclient.ConnectionStatus, mr metricsCollectionStats) time.Duration {
+	// TODO: passing the whole mr struct looks overkilling, given we only want a single value, the average, and maybe
+	//  we can actually use a version of the algorithm that calculates it (and the number of items)
+	var nItems int
 	var sd float64
-
 	for _, item := range gc {
 		if item.GetConnectionStatus() == status {
+			nItems++
 			sd += math.Pow(float64(item.GetTCPProcessingDuration(status))-float64(mr.avg), 2)
 		}
 	}
-	return time.Duration(math.Sqrt(sd / float64(time.Duration(mr.numberOfConnections))))
+	if nItems == 0 {
```

Comment:
did you find this while testing it? 😃

Reply:
why nItems instead of numberOfConnections?

Reply:
I want to decouple the stdDev function as much as possible from our structures and "external requirements": i really feel making it autonomous will bring clarity...

Reply:
👍